source/functional-concepts.md

```diff
@@ -0,0 +1,150 @@
+# Functional Programming in Swift: A Primer
+
+Swift is a hybrid language that allows both objective-oriented and functional code to co-exist.
+
+### First-class functions
+
+First-class functions are *values* that can either take other functions as arguments or return them as results. Also refered to as *higher order functions*.
```

Comment:
A first-class function is a function that can be treated as a value. A higher-order function is a function that operates on functions (takes a function as argument, returns a function).